graphistry/umap_utils.py

```diff
@@ -492,6 +492,12 @@ def umap(
             logger.debug("umap X_: %s", X_)
             logger.debug("umap y_: %s", y_)
 
+            if isinstance(X_,pd.DataFrame):
+                index_to_nodes_dict = dict(zip(range(len(nodes)), nodes))
+            elif isinstance(X_,cudf.DataFrame):
+                index_to_nodes_dict=cudf.DataFrame(nodes).reset_index()
+                X_=pd.DataFrame(X_.to_numpy())
```

Comment:
shouldn't we keep X_ as a GPU df, and enable process_umap etc handle it? when in gpu mode, generally easier to keep everything as gpu obj and only temporarily back out to cpu for calls that can't... a lot less to keep track of

Reply:
handling X_ properly is last task